### ISSUE TITLE
GetScript(s)<T>() should support interfaces

### ISSUE THREAD

**Issue by honzapatCZ** [FlaxEngine/FlaxEngine#290]
Example extension in c#, but prob. should be c++ level

```cs
public static T GetScriptInterface<T>(this Actor actor)
    {
        for (int i = 0; i < actor.ScriptsCount; i++)
        {
            if (actor.GetScript(i) is T pettable)
            {
                return pettable;
            }
        }
        return default(T);
    }
    public static T[] GetScriptInterfaces<T>(this Actor actor)
    {
        List<T> list = new List<T>();
        for (int i = 0; i < actor.ScriptsCount; i++)
        {
            if (actor.GetScript(i) is T pettable)
            {
                list.Add(pettable);
            }
        }
        return list.ToArray();
    }
```

**Comment by kerelape** [comment]
any updates on this?

**Comment by mafiesto4** [comment]
Implemented in d58a77cc60a0447a3b36079efc5797053231e27e.
